### DISCUSSION TITLE
Camera fully functionnal but restarts very often

### DISCUSSION THREAD

**Discussion by cnico** [roleoroleo/yi-hack-Allwinner-v2#334] [Q&A] [answered]
Hi,

I have a Yi Dome 1080p with a serial starting by IFUS with a firmware version in 9.0.05*. So I installed the r30gb file in version 0.1.9.
It works successfully : the camera is hacked and I have access to the web interface and all the functionnalities including PTZ. I integrated it in homeassistant with no problem :-).
I know it is a beta version but for me It works great.
The only drawback is that the camera restarts very often. In a typical day, it restarts at least every one or two hour and sometimes it restarts 3 times in a row every 5 minutes. I suspect it could come from the day/night switching context when it occurs at 5 min interval but i am not sure if it is the root cause.

How could I diagnose it by connecting through SSH I suppose, but then ? Are there log files that could help identify what causes the camera to restart ? Are there tools to run to get debug informations after a restart ?

Thank you for the work on the hack and for your help,

**Comment by roleoroleo** [comment, marked as answer]
Did you enable swap space?
If not, when the cam take a snapshot, the memory is not enough and the cam dies.

**Reply by cnico** [reply]
Thank you for the answer : I enabled it and the camera did not reboot during last day.
A small improvement of the hack could be to activate it by default (I do not remember to have deactivated it when I installed the camera).
